packages/v4/src/content/design-guidelines/components/menu/menu.md

```diff
@@ -0,0 +1,82 @@
+
+<img src="./img/option-select-menu.png" alt="option select menus" width="863"/>
+
+1. **Selected item:** A checkmark to the right of a menu item indicates that it is currently selected.
+
+2. **Icon (optional):** You may place an icon before each menu item to aid in recognition. It has been shown that familiar icons speed the recognition of text labels, but only use icons if they will be easily recognized and are distinguishable. Never use icons simply as a decoration.
+
+3. **Descriptions (optional):** Descriptive text may be added below each menu item. This should only be used when the menu item label by itself may not be clear to all users. Keep descriptive text to two lines or less.
+
+### Grouped menus
+You may decide to group menu items to associate related items and/or to indicate a hierarchy of items. Items may be grouped using group headings, separators, or both.
+
+<img src="./img/grouped-menu.png" alt="grouped menu" width="348"/>
+
+1. **Group heading (optional):** Add a group heading when you want to name the group.
+
+2. **Separator:** Separators are horizontal dividers that help to group the menu items by clearly showing where one group ends and the next begins.
+
+### Multi-level fly-out menus
+Fly-out menus can be used when you want to expose sub-items from a parent node. PatternFly supports a single level fly-out, only.
+
+<img src="./img/flyout-menu.png" alt="flyout menu" width="348"/>
+
+If a menu item has sub-items, a “>” character will be shown to the right of the item label. When hovering over the item, the secondary menu will be exposed.
+
+### Filtering menu items
+If the list of possible items is very long, a filter may be added to make it easier for the user to find what they are looking for.
+
+<img src="./img/menu-filtering.png" alt="grouped menu" width="348"/>
+
+When filtering is being used, a search input will be added to the top of the menu and the list of items will be filtered as the user types.
+
+### Favoriting and other actions
+Favoriting or any other optional action can be associated with any menu item. In this case, making an item a favorite will duplicate in at the top of the menu. This may be useful when you have a long list of possible actions and want to “pin” a subset of items to the top of the menu.
+
+<img src="./img/favorities-menu.png" alt="menu with favorites" width="348"/>
+
+
+Any action that can be represented as an icon button can be placed in one or more menu items. In this case the favorites action is used to mark an item as a favorite.
+
+## Usage
+
+| **When you want to...** |                     |
+| ------------------- | ------------------- |
+| Expose a list of actions in a dropdown menu | Use an actions menu with actions and/or links. Add icons in front of menu items only if they will help to make items more recognizable. Include descriptive text for menu items that may require further definition. |
```

Comment:
would take out the ellipses at the end of each use case, the first row doesn't have them, and some are alone on the next line which looks a little weird